### PR TITLE
Date time function

### PR DESCRIPTION
**Pull request by neilpeterson**
https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-date#datetimeadd

Fixes https://github.com/microsoft/vscode-azurearmtools/issues/636

### CODE REVIEW

File: assets/ExpressionMetadata.json [hunk 1]
```diff
@@ -99,6 +99,13 @@
             "minimumArguments": 1,
             "maximumArguments": 1
         },
+        {
+            "name": "dateTimeAdd",
+            "expectedUsage": "dateTimeAdd(base, duration, [format])",
+            "description": "Adds a time duration to a base datetime value.",
```
Comment (bergmeister) [inline assets/ExpressionMetadata.json:105]: The type is actually string and not datetime
```suggestion
            "description": "Adds a time duration to a base value. ISO 8601 format is expected.",
```

Comment (neilpeterson) [reply assets/ExpressionMetadata.json:105]: @alex-frankel can you advise? I've taken the description verbatim from docs - thanks.

https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-date#datetimeadd

Comment (alex-frankel) [reply assets/ExpressionMetadata.json:105]: Yeah I guess datetime could be read as a type, when the intention was a format for the string. This looks good. @tfitzmac FYI - we should probably update the doc as well

Comment (tfitzmac) [reply assets/ExpressionMetadata.json:105]: Thanks, I'll update the doc today.

**Comment by StephenWeatherford** [comment]
@neilpeterson Not sure if the above issues are resolved - merge whenever you're ready.

**Comment by bergmeister** [comment]
Awesome, when do you plan to release this change?

**Comment by neilpeterson** [comment]
This update will be in 0.10.0, rough release estimates are here:

https://github.com/microsoft/vscode-azurearmtools/milestones

**Pull request merged**